Graphs/Dijikstra.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+This program is for dijkestra
+It finds the shortest path from 1 to n (can be easily changed)
+It also returns the path
+*/
+
+#include <iostream>
+#include <queue>
+#define INF 1e17
+
+using namespace std;
+
+typedef pair<long long, long long> pii;
+
+vector<int> Dijkstra(const vector<vector<pii>> &graph, int src, int dest) {
+	int n = graph.size();
+	n--;
+	vector<long long> dist(n + 1, INF); //cost, vertex
+	vector<bool> visited(n + 1, false);
+	priority_queue<pii, vector<pii>, greater<pii>> pq; 
+
+	dist[src] = 0;
+	for (int i = 1; i <= n; i++) {
+		pq.push({dist[i], i});
+	}
+	
+	vector<int> parent(n+1);
+	parent[src] = src;
+
+	while(!pq.empty()) {
+		pii top = pq.top();
+		pq.pop();
+		long long current_cost = top.first;
+		int current_vertex = top.second;
+
+		if (visited[current_vertex]) {
+			continue;
+		}
+
+		visited[current_vertex] = true;
+
+		for (auto neighbor: graph[current_vertex]) {
+			int neighbor_vertex = neighbor.first;
+			long long edge = neighbor.second;
+
+			if (visited[neighbor_vertex]) {
```

Comment:
You don't need this condition, because if it's has been visited that means that dijkstra cannot find a better cost and hence would never be pushed again.